Require less in saas-6 and extract theme_default from website

dmo-odoo opened the pull request:
AL wants to make those two changes to saas-6 before saas customers are migrated to saas-6.
@KangOl I'd like your thoughts on the subject.

Of course it needs to be squashed into two commits, I just wanted to keep al's first work separated until this is approved.

KangOl commented on addons/website/views/website_templates.xml:
```diff
@@ -461,54 +463,57 @@
     </ul>
 </template>
 
-<template id="kanban">
```
Where does this view gone?

dmo-odoo replied:
Lingering change from al's commits, I need to check with him.

KangOl commented:
Please keep change minimal: reorder of templates is unneeded and make diff difficult to read.

Also what the point of extracting the theme in its own module? does it works if not installed?

dmo-odoo commented:
I think the reordering of the templates is needed for the website_templates.xml file, it is a real mess right now. I understand it makes the diff difficult to read though.

The reason why we extracted the theme in its own module is quite simple. Enabling views from this "default theme" conflicts with other themes. I solved it for our themes by disabling those default views in theme_common. However, al didn't want that community theme developers have to think about that and also disable those views in every theme. Extracting them in their own theme will allow us to handle its uninstallation the same way we will handle the uninstallation of other themes to avoid conflict on the saas.

This part is not done already, but it is planned to add it as it will be required for saas customers who will indeed try to switch theme without uninstalling the previous one...

dmo-odoo commented:
@KangOl I squashed and cleaned the commits, please have a look. If the reordering of the template is really bothering you, you can look at the separate commits. There is not that much actually, most of this PR is displacement and renaming of small stuffs like inherit_ids.
